main.py

```diff
@@ -154,6 +164,52 @@ def workshop(workshop):
     return render_template("workshop.html", **data)
 
 
+class User:
+    def __init__(self, Id, username, password):
+        self.id = Id
+        self.username = username
+        self.password = password
+
+
+users = []
+users.append(User(Id=1, username="admin", password="admin"))
+users.append(User(Id=2, username="Becca", password="secret"))
+
+
+@app.route("/logout.html")
+# @login_required
+def logout():
+    session.clear()
+    flash("Successfully logged out")
+    return redirect(url_for("index"))
+
+
+@app.route("/login.html", methods=["GET", "POST"])
+def login():
+    error = ""
+    try:
+        if request.method == "POST":
```

Comment:
Unfortunately, I don't think any of this code will run in the frozen static version of the site. 

Maybe that is okay though if it is for different live version.

Reply:
+1 I agree it would be nice if we can make this work on static website!
It seems that AWS has some nice solution [here](https://douglasduhaime.com/posts/s3-lambda-auth.html). But I'm not sure if this main repo needs to be tied with AWS.